src/mon/Monitor.cc

```diff
@@ -5576,14 +5575,17 @@ int Monitor::scrub_start()
   dout(10) << __func__ << dendl;
   ceph_assert(is_leader());
 
-  if (!scrub_result.empty()) {
-    clog->info() << "scrub already in progress";
-    return -EBUSY;
+  if (std::shared_ptr<ScrubContext> local_ctx = scrub_con.load();  scrub_con.load() !=
```

Comment:
There are 2 loads from `scrub_con`; they can race. I think the intention was to compare just `local_ctx`:

```suggestion
if (std::shared_ptr<ScrubContext> local_ctx = scrub_con.load();  local_ctx) {
```

Reply:
done